src/testplan/testplanTestRunner.ts

```diff
@@ -0,0 +1,203 @@
+            `--xml=${dirName}`
+        ].concat(runArguments.argumentsToPass);
+        this.logger.log('info', `Running testplan with arguments: ${testRunArguments.join(', ')}`);
+
+        const testExecution = this.runTestPlan(config, additionalEnvironment, testRunArguments);
+
+        this.testExecutions.set(test, testExecution);
+        await testExecution.complete();
+        this.testExecutions.delete(test);
+
+        this.logger.log('info', 'Test execution completed');
+        const states = await parseTestStates(dirName);
+
+        cleanupCallback();
+        return states;
+    }
+
+    private runTestPlan(config: IWorkspaceConfiguration, env: IEnvironmentVariables, args: string[])
+    : IProcessExecution
+    {
+        const testplanPath = config.getTestplanConfiguration().testplanPath();
+
+        this.logger.log('info', `Running ${testplanPath} as an executable`);
+        return runProcess(
+            config.pythonPath(),
+            [testplanPath].concat(args),
+            {
+                cwd: config.getCwd(),
+                environment: env,
+                acceptedExitCodes: TESTPLAN_NON_ERROR_EXIT_CODES,
+            });
+    }
+
+    private async loadEnvironmentVariables(config: IWorkspaceConfiguration): Promise<IEnvironmentVariables> {
+        const envFileEnvironment = await EnvironmentVariablesLoader.load(config.envFile(), process.env, this.logger);
+
+        const updatedPythonPath = [
+            config.getCwd(),
+            envFileEnvironment.PYTHONPATH,
+            process.env.PYTHONPATH
+        ].filter(item => item).join(path.delimiter);
+
+        return {
+            ...envFileEnvironment,
+            PYTHONPATH: updatedPythonPath,
+            TESTPLAN_PLUGINS: envFileEnvironment.TESTPLAN_PLUGINS,
+        };
+    }
+
+    private async getJunitReportPath(
+        cwd: string,
+        runArguments: IRunArguments
+    ): Promise<{ dirName: string, cleanupCallback: () => void }> {
+        if (runArguments.junitReportPath) {
+            return Promise.resolve({
+                dirName: path.resolve(cwd, runArguments.junitReportPath),
+                cleanupCallback: () => { /* intentionally empty */ },
+            });
+        }
+        return await this.createTemporaryDirectory();
+    }
+
+    private getDiscoveryArguments(rawTestplanArguments: string[]): string[] {
+        const argumentParser = this.configureCommonArgumentParser();
+        const [, argumentsToPass] = argumentParser.parse_known_args(rawTestplanArguments);
+        return ['--info', 'pattern-full'].concat(argumentsToPass);
+    }
+
+    private getRunArguments(test: string, rawTestplanArguments: string[]): IRunArguments {
+        const argumentParser = this.configureCommonArgumentParser();
+
+        const [knownArguments, argumentsToPass] = argumentParser.parse_known_args(rawTestplanArguments);
+        return {
+            junitReportPath: (knownArguments as { xmlpath?: string }).xmlpath,
```

Comment:
I don't see `xmlpath` in configured arguments, so I assume it's always undefined?

Reply:
true, I added it to argsParser, so people can redefine its value